classes/publication/DAO.php

```diff
@@ -441,36 +441,30 @@ protected function deleteControlledVocab(int $publicationId)
     /**
      * Set a publication's category property
      */
-    protected function setCategories(Publication $publication)
+    protected function setCategories(Publication $publication): void
     {
-        $publication->setData(
-            'categoryIds',
-            Repo::category()->getCollector()
-                ->filterByPublicationIds([$publication->getId()])
-                ->getIds()
-                ->toArray()
-        );
+        $categoryIds = PublicationCategory::withPublicationId($publication->getId())->pluck('category_id');
+        $publication->setData('categoryIds', $categoryIds);
     }
 
     /**
      * Save the assigned categories
      */
-    protected function saveCategories(Publication $publication)
+    protected function saveCategories(Publication $publication): void
     {
-        Repo::category()->dao->deletePublicationAssignments($publication->getId());
-        if (!empty($publication->getData('categoryIds'))) {
-            foreach ($publication->getData('categoryIds') as $categoryId) {
-                Repo::category()->dao->insertPublicationAssignment($categoryId, $publication->getId());
-            }
+        $categoryIds = (array) $publication->getData('categoryIds');
+        if (empty($categoryIds)) {
+            $categoryIds = [];
```

Comment:
As you're casting to array in top, this isn't needed, it will be always a filled/empty array.